src/features/feature.ts

```diff
@@ -42,9 +42,9 @@ export type Replace<T, S> = S & Omit<T, keyof S>;
  * with "this" bound to what the type will eventually be processed into.
  * Intended for making lazily evaluated objects.
  */
-export type OptionsFunc<T, R = Record<string, unknown>, S = R> = () => T &
-    Partial<R> &
-    ThisType<T & S>;
+export type OptionsFunc<T, R = Record<string, unknown>, S = R> = () => OptionsObject<T,R,S>;
+
+export type OptionsObject<T, R = Record<string, unknown>, S = R> = T & Partial<R> & ThisType<T & S>;
```

Comment:
What is the purpose in splitting this type up like that?

Reply:
This was because I was expecting the decorators to need to know the typing of the objects that were coming in, but I *think* this could be safely undone, and the decorators forcing additional typing on the `OptionFunc`s would still work.

Reply:
Ok, on further efforts, this does need to be split - the decorators do need access to the OptionsFunc, after it's been called, in order to have the decorator's typing merged into the `createFeature` function. Without this, you just get mismatched type errors upon trying to use *any* decorator at all, only fixable by manually applying the enhanced type every time it's used - which completely bypasses the purpose of these.